packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,572 @@
+  /**
+   * Whether the item is expanded
+   */
+  isExpanded?: boolean;
+  /**
+   * Whether the item is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether the item is active/selected
+   */
+  isSelected?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(event: MouseEvent<HTMLAnchorElement>): void;
+} & (
+  | {
+      /**
+       * Determines if the side nav item is a section header
+       */
+      isSectionHeader: true;
+      href?: never;
+      children?: never;
+    }
+  | {
+      isSectionHeader?: false;
+      href: string;
+      /**
+       * An array of side nav items
+       */
+      children?: SideNavItem[];
+    }
+);
+
+export type SideNavFooterItem = {
+  id: string;
+  label: string;
+  href: string;
+};
+
+const SideNavCollapsedContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "sideNavCollapsed",
+})(
+  ({
+    odysseyDesignTokens,
+    sideNavCollapsed,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    sideNavCollapsed: boolean;
+  }) => ({
+    backgroundColor: odysseyDesignTokens.HueNeutral300,
+    paddingTop: odysseyDesignTokens.Spacing5,
+    cursor: "pointer",
+    width: sideNavCollapsed ? "auto" : 0,
+    visibility: sideNavCollapsed ? "visible" : "hidden",
+  }),
+);
+
+const SideNavContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "sideNavCollapsed",
+})(
+  ({
+    odysseyDesignTokens,
+    sideNavCollapsed,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    sideNavCollapsed: boolean;
+  }) => ({
+    backgroundColor: odysseyDesignTokens.HueNeutralWhite,
+    flexDirection: "column",
+    display: "flex",
+    visibility: sideNavCollapsed ? "hidden" : "visible",
+    width: sideNavCollapsed ? "0" : "100%",
+    transitionProperty: "width, visibility",
+    transitionDuration: odysseyDesignTokens.TransitionDurationMain,
+    transitionTimingFunction: odysseyDesignTokens.TransitionTimingMain,
+  }),
+);
+
+const SideNavHeaderContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  display: "flex",
+  justifyContent: "space-between",
+  alignItems: "center",
+  paddingLeft: odysseyDesignTokens.Spacing4,
+  paddingRight: odysseyDesignTokens.Spacing4,
+  paddingTop: odysseyDesignTokens.Spacing3,
+  paddingBottom: odysseyDesignTokens.Spacing3,
+}));
+
+const CollapseIcon = ({ onClick }: { onClick?(): void }): ReactElement => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  return (
+    <div
+      tabIndex={0}
+      role="button"
+      onClick={onClick}
+      onKeyDown={(event) => {
+        event.key === "Enter" && onClick && onClick();
+      }}
+    >
+      <Box
+        component="a"
+        sx={{
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+          width: "32px",
```

Comment:
Since the icon already has a set size, we can avoid setting an explicit width/height and just add padding. (`odysseyDesignTokens.Spacing4` is 16px)

Reply:
refactored and removed the Box